metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -887,28 +898,29 @@ def _init_tableau_project_registry(self, all_project_map: dict) -> None:
             logger.debug(f"Project {project.name} is added in project registry")
             projects_to_ingest[project.id] = project
 
-        # We rely on automatic browse paths (v2) when creating containers. That's why we need to sort the projects here.
-        # Otherwise, nested projects will not have the correct browse paths if not created in correct order / hierarchy.
-        self.tableau_project_registry = OrderedDict(
-            sorted(projects_to_ingest.items(), key=lambda item: len(item[1].path))
-        )
-
         if self.config.extract_project_hierarchy is False:
             logger.debug(
                 "Skipping project hierarchy processing as configuration extract_project_hierarchy is "
                 "disabled"
             )
-            return
+        else:
+            logger.debug(
+                "Reevaluating projects as extract_project_hierarchy is enabled"
+            )
 
-        logger.debug("Reevaluating projects as extract_project_hierarchy is enabled")
+            for project in list_of_skip_projects:
+                if (
+                    project.parent_id in projects_to_ingest
+                    and self._is_denied_project(project) is False
+                ):
+                    logger.debug(f"Project {project.name} is added in project registry")
+                    projects_to_ingest[project.id] = project
 
-        for project in list_of_skip_projects:
-            if (
-                project.parent_id in self.tableau_project_registry
-                and self._is_denied_project(project) is False
-            ):
-                logger.debug(f"Project {project.name} is added in project registry")
-                self.tableau_project_registry[project.id] = project
+        # We rely on automatic browse paths (v2) when creating containers. That's why we need to sort the projects here.
```

Comment:
I reverted the changes to include the extract_project_hierarchy part again, as discussed. However, I slightly refactored it so we do the sorting below at the very end for all the projects to ingest.